performance-measure/src/main/java/org/sonarsource/performance/measure/DurationMeasureFiles.java

```diff
@@ -0,0 +1,250 @@
+  public static DurationMeasure fromJson(JsonObject jsonObject) {
+    Map<String, DurationMeasure> childrenMap = null;
+    JsonArray children = jsonObject.getAsJsonArray("children");
+    if (children != null && children.size() > 0) {
+      childrenMap = new HashMap<>();
+      for (JsonElement child : children) {
+        DurationMeasure childMeasure = fromJson(child.getAsJsonObject());
+        childrenMap.merge(childMeasure.name(), childMeasure, DurationMeasure::merge);
+      }
+    }
+    String name = jsonObject.getAsJsonPrimitive("name").getAsString();
+    long calls = jsonObject.getAsJsonPrimitive("calls").getAsLong();
+    long durationNanos = jsonObject.getAsJsonPrimitive("durationNanos").getAsLong();
+    return new DurationMeasure(name, calls, durationNanos, childrenMap);
+  }
+
+  /**
+   * @param path the destination statistics file
+   * @param measure the root of the measure hierarchy
+   * @param categoryNames Map defining categories for grouped measures, when a measure.name() matches
+   *                      a key of this map, the value of the map become the category of the measure
+   *                      and its descendents.
+   * @param groupedMeasurePredicate Predicate applied on each measure.name() to decide if the measure
+   *                                need to be extracted, grouped and categorised at the end of the
+   *                                statistics file.
+   */
+  public static void writeStatistics(Path path, DurationMeasure measure, Map<String, String> categoryNames,
+    Predicate<String> groupedMeasurePredicate) throws IOException {
+    Files.write(path, toStatistics(measure, categoryNames, groupedMeasurePredicate).getBytes(UTF_8));
+  }
+
+  public static String toStatistics(DurationMeasure measure, Map<String, String> categoryNames, Predicate<String> groupedMeasurePredicate) {
+    Map<String, Set<String>> measureCategories = categorizeMeasures(measure, categoryNames, Collections.emptySet());
+    DurationMeasure measureWithoutGrouped = measure.copy();
+    measureWithoutGrouped.remove(MeasurementCost.MEASUREMENT_COST_NAME);
+    measureWithoutGrouped.remove(MeasurementCost.SUBTRACTED_MEASUREMENT_COST_NAME);
+    Map<String, DurationMeasure> groupedMeasure = extractGroupedMeasure(measureWithoutGrouped, groupedMeasurePredicate);
+    StringBuilder stat = new StringBuilder();
+    stat.append("Performance (in seconds without observation cost)\n");
+    stat.append(toTextTree(measureWithoutGrouped));
+    stat.append("\n");
+    if (!groupedMeasure.isEmpty()) {
+      stat.append("Grouped Entries (in seconds without observation cost)\n");
+      stat.append(toRankedList(groupedMeasure, measureCategories));
+    }
+    return stat.toString();
+  }
+
+  /**
+   * Walk recursively through the measure hierarchy and assign a list of category to some measure names.
+   * If a measure.name() is a key in "categoryNames", then the related value in the "categoryNames" map become the measure category.
```

Comment:
become -> becomes

Reply:
I should definitely find a Grammarly plugin for my IDE...